lib/ash/actions/read/relationships.ex

```diff
@@ -165,33 +165,50 @@ defmodule Ash.Actions.Read.Relationships do
             |> Map.put(:api, join_relationship.api || related_query.api)
             |> hydrate_refs(relationship.source)
 
-          case through_query.api.can(through_query,
-                 alter_source?: true,
-                 run_queries?: false,
-                 base_query: through_query
-               ) do
-            {:ok, true} ->
-              {:ok,
-               [
-                 {clear_lateral_join_source(source_query), relationship.source_attribute,
-                  relationship.source_attribute_on_join_resource, relationship},
-                 {clear_lateral_join_source(through_query),
-                  relationship.destination_attribute_on_join_resource,
-                  relationship.destination_attribute, join_relationship}
-               ]}
-
-            {:ok, true, authorized_through_query} ->
-              {:ok,
-               [
-                 {clear_lateral_join_source(source_query), relationship.source_attribute,
-                  relationship.source_attribute_on_join_resource, relationship},
-                 {clear_lateral_join_source(authorized_through_query),
-                  relationship.destination_attribute_on_join_resource,
-                  relationship.destination_attribute, join_relationship}
-               ]}
+          if source_query.context[:private][:authorize?] do
```

Comment:
It feels like I'm adding a lot of `if` statements recently, but I was unable to come up with something better in the short term.

Do you have some rule of thumb for things like this? Would you refactor some things out into smaller functions to circumvent this or is there another way?

Reply:
Yeah, if there is a need to refactor I'd focus on extracting something out into a named function. Often I don't end up doing this, much to the chagrin of others. Feel free to do so if you like :)

Reply:
I'm ok with merging it like this for now, I still have a couple other tests that fail with the new read action code. That has priority for me right now.